### ISSUE TITLE
G002 gives a false positive with translated strings

### ISSUE THREAD

**Issue by jpmckinney** [astral-sh/ruff#13350]
For example, when using gettext, it's common to write code like:

```python
_("Loading module: %(module)s") % {"module": mod}
```

or:

```python
_("Loading module: %s") % mod
```

If these messages are logged (for example, in a multilingual command line tool), like:

```python
logger.info(_("Loading module: %s") % mod)
```

Then, G002 is raised, although the `%` operation is on the return value of a function (which is not how people typically misuse the logging module).

**Comment by MichaReiser** [comment]
I feel hesitant to allow the format operator when the left side is a call expression. For example:

```python
```
>>> def test(): return "abcd %s"
... 
>>> logger.info(test() % "1")
```

This could be rewritten to 

```python
def logTet(arg):
	logger.info("test %", arg)

log(1)
```

I also think flagging the `gettext` call is somewhat in the spirit of the rule. Ideally, `gettext` only gets called if the `logger`'s level is higher than the called log function to avoid an unnecessary gettext calls and string interpolations. 

I would probably explore adding custom `infoTranslated` helper methods that conditionally call `gettext`.

**Comment by jpmckinney** [comment]
Ahh, good call. I'll close – but if some future user does want something similar, it can of course be reopened.
